arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1064,8 +1075,26 @@ Result IResearchDataStore::commitUnsafeImpl(
     }
 
     // update reader
-    TRI_ASSERT(_dataStore._reader != reader);
-    _dataStore._reader = reader;
+    TRI_ASSERT(_dataStore.loadSnapshot()->_reader != reader);
+    auto const readerSize = reader->size();
+    auto const docsCount = reader->docs_count();
+    auto const liveDocsCount = reader->live_docs_count();
+
+    // For now we want to retain old behaviour that
+    // storage snapshot is not older than iresearch snapshot
+    // TODO: Remove this as soon as index_writer will start accepting limiting
+    // tick for commit
+    auto engineSnapshotAfterCommit = _engine->currentSnapshot();
+    if (ADB_UNLIKELY(!engineSnapshotAfterCommit)) {
+      return {
+          TRI_ERROR_INTERNAL,
+          absl::StrCat("Failed to get engine snapshot while finishig commit "
+                       "ArangoSearch index '",
+                       index().id().id(), "'")};
+    }
+
+    _dataStore.storeSnapshot(std::make_shared<DataSnapshot>(
```

Comment:
This can also throw. Do we need to roll back then?

Reply:
I guess no. In the worst case we will have outdated snapshot like commit has failed. And will try to update  it as soon as next changes will arrive.